### PR TITLE
Fix clicking when there is no active cursor.

### PR DESCRIPTION
**Pull request by mixonic**
Also refactors the `offsets` return value into a `Range` object consisting of two `Position` objects. These encapsulate the difficulties of creating a meaningful position data-set for a given DOM node.

### CODE REVIEW

**Comment by mixonic** [comment]
Also moved `cursor` into `utils`. I'd like to get to a point where the `models` directory is all abstract post stuff.

File: src/js/utils/cursor.js [hunk 1]
```diff
@@ -98,87 +98,29 @@ export default class Cursor {
 
   get offsets() {
     const { sections } = this.post;
-    const selection = this.selection;
-    const { isCollapsed, rangeCount } = selection;
-    const range = rangeCount > 0 && selection.getRangeAt(0);
+    const { selection } = this;
 
-    if (!range) {
+    if (selection.rangeCount === 0 || !selection.getRangeAt(0)) {
```
Comment (bantic) [inline src/js/utils/cursor.js]: I think `selection.rangeCount === 0` is sufficient. I'm not sure what circumstance would result in the rangeCount being non-zero and `getRangeAt(0)` returning `undefined`

File: src/js/utils/cursor/position.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        offsetInSection
+      );
+      marker = markerPosition.marker;
+      offsetInMarker = markerPosition.offset;
+    }
+
+    this.section = section;
+    this.offsetInSection = offsetInSection;
+    this.marker = marker;
+    this.offsetInMarker = offsetInMarker;
```
Comment (bantic) [inline src/js/utils/cursor/position.js]: seems like we could use getters to make these calculations lazy

File: tests/acceptance/basic-editor-test.js [hunk 3]
```diff
@@ -62,3 +57,27 @@ test('#disableEditing and #enableEditing toggle contenteditable', (assert) => {
...
 });
+
+test('clicking outside the editor does not raise an error', (assert) => {
+  editor = new Editor({autofocus: false});
+  editor.render(editorElement);
+
+  let secondEditorElement = document.createElement('div');
+  document.body.appendChild(secondEditorElement);
+
+  let secondEditor = new Editor(); // This editor will be focused
```
Comment (bantic) [inline tests/acceptance/basic-editor-test.js]: could make the comment redundant by passing `{autofocus:true}`

**Comment by bantic** [comment]
:+1:

**Pull request merged**